File: src/fiefapp.svelte/src/models/boatbuilder.ts
import type { IPerson } from './person';
import type { Shipyard } from './shipyard';

interface IBoatbuilder {
  id?: string;
  shipyard?: Shipyard;
}

export class Boatbuilder implements IBoatbuilder, IPerson {
  id: string;
  boatId?: string;
  shipyard?: Shipyard;
  name: string;
  resources: number;
  skill: number;
  loyalty: number;
  age: number;

  constructor() {
    this.id = '';
    this.boatId = undefined;
    this.shipyard = undefined;
    this.name = '';
    this.resources = 0;
    this.skill = 0;
    this.loyalty = 0;
    this.age = 0;
  }
}

File: src/fiefapp.entities/Boatbuilder.cs
using fiefapp.entities.BaseClasses;
using fiefapp.entities.Interfaces;
using System;
using System.Collections.Generic;

namespace fiefapp.entities
{
    public class Boatbuilder : Person, IBaseEntity
    {
        public Boatbuilder()
        {
            Id = Guid.NewGuid();
            Notes = new HashSet<string>();
        }

        public override Guid Id { get; set; }
        public override string Name { get; set; }
        public override int Age { get; set; }
        public override int Resources { get; set; }
        public override int Loyalty { get; set; }
        public override int Skill { get; set; }
        public override ICollection<string> Notes { get; set; }
        public Guid? ShipyardId { get; set; }
        public Shipyard Shipyard { get; set; }
    }
}
